### PR TITLE
Fixes #14 and #15: Imports SQS and checks twilio error code

### LINKED ISSUE #14: Missing SQS import
This is on prod-nys-sms:

```
2020-08-28T16:42:36.837-04:00

Copy
2020-08-28T20:42:36.837Z	6698b7e0-6129-5ec8-89aa-e0230f5a1213	ERROR	Invoke Error	
{
    "errorType": "ReferenceError",
    "errorMessage": "sqs is not defined",
    "stack": [
        "ReferenceError: sqs is not defined",
        "    at Runtime.exports.handler (/var/task/sms.js:47:7)",
        "    at process._tickCallback (internal/process/next_tick.js:68:7)"
    ]
}
```

### PR DESCRIPTION
**Pull request by dennisgove**
For #14: I've added the proper SQS import and setup code
For #15: I've added a check against the error code. If the code is 21211 (twilio invalid phone code) then we will log out at INFO level "twilio rejected mobile as invalid - sms not sent for verification code 12345 and will not be retried" (with the appropriate verification code.

@segfault for #14 I've not been able to confirm completely that the message is added back to the queue. It appears that the lambda doesn't have the appropriate permissions to post to the queue (or the `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` parameters are not what I'm expecting them to be). I'm still seeing error 

```
{
    "errorType": "InvalidClientTokenId",
    "errorMessage": "The security token included in the request is invalid.",
    "code": "InvalidClientTokenId",
    "message": "The security token included in the request is invalid.",
    "time": "2020-08-30T01:00:42.955Z",
    "requestId": "f92d1879-9fd1-575c-aba7-4fbd63abd940",
    "statusCode": 403,
    "retryable": false,
    "retryDelay": 61.191860126264494,
    "stack": [
        "InvalidClientTokenId: The security token included in the request is invalid.",
        "    at Request.extractError (/var/task/node_modules/aws-sdk/lib/protocol/query.js:50:29)",
        "    at Request.callListeners (/var/task/node_modules/aws-sdk/lib/sequential_executor.js:106:20)",
        "    at Request.emit (/var/task/node_modules/aws-sdk/lib/sequential_executor.js:78:10)",
        "    at Request.emit (/var/task/node_modules/aws-sdk/lib/request.js:683:14)",
        "    at Request.transition (/var/task/node_modules/aws-sdk/lib/request.js:22:10)",
        "    at AcceptorStateMachine.runTo (/var/task/node_modules/aws-sdk/lib/state_machine.js:14:12)",
        "    at /var/task/node_modules/aws-sdk/lib/state_machine.js:26:10",
        "    at Request.<anonymous> (/var/task/node_modules/aws-sdk/lib/request.js:38:9)",
        "    at Request.<anonymous> (/var/task/node_modules/aws-sdk/lib/request.js:685:12)",
        "    at Request.callListeners (/var/task/node_modules/aws-sdk/lib/sequential_executor.js:116:18)"
    ]
}
```

### CODE REVIEW

**Comment by aisegfault** [comment]
looks good

File: sms.js [hunk 1]
```diff
@@ -14,6 +15,11 @@ exports.handler = async function(event) {
     queueUrl,
     smsTemplate
   } = await getSmsConfig()
+  const sqs = new SQS({
+    accessKeyId: process.env.AWS_ACCESS_KEY_ID,
```
Comment (segfault) [inline sms.js]: Running in the lambda environment the api will discover this by looking elsewhere, these shouldn't be passed.

**Review by segfault — CHANGES_REQUESTED** [review]
LGTM w/ the removal of the env usage. This needs a matching change in the terraform so that the `sqs:SendMessage` priv is granted to this lambda.

**Comment by dennisgove** [comment]
Removed accessKey stuff from SQS initialization. Am now getting the following error, which I imagine will go away when proper policies are setup.

```
{
  "errorType": "AccessDenied",
  "errorMessage": "Access to the resource https://sqs.us-east-1.amazonaws.com/ is denied.",
  "trace": [
    "AccessDenied: Access to the resource https://sqs.us-east-1.amazonaws.com/ is denied.",
    "    at Request.extractError (/var/task/node_modules/aws-sdk/lib/protocol/query.js:50:29)",
    "    at Request.callListeners (/var/task/node_modules/aws-sdk/lib/sequential_executor.js:106:20)",
    "    at Request.emit (/var/task/node_modules/aws-sdk/lib/sequential_executor.js:78:10)",
    "    at Request.emit (/var/task/node_modules/aws-sdk/lib/request.js:683:14)",
    "    at Request.transition (/var/task/node_modules/aws-sdk/lib/request.js:22:10)",
    "    at AcceptorStateMachine.runTo (/var/task/node_modules/aws-sdk/lib/state_machine.js:14:12)",
    "    at /var/task/node_modules/aws-sdk/lib/state_machine.js:26:10",
    "    at Request.<anonymous> (/var/task/node_modules/aws-sdk/lib/request.js:38:9)",
    "    at Request.<anonymous> (/var/task/node_modules/aws-sdk/lib/request.js:685:12)",
    "    at Request.callListeners (/var/task/node_modules/aws-sdk/lib/sequential_executor.js:116:18)"
  ]
}
```

**Comment by dennisgove** [comment]
While testing this I did see that Twilio is no longer authenticating us which prevents us from sending SMS messages to even valid mobile numbers (tested with my mobile). Last night it was sending messages to my phone so something appears to have changed somewhere.

```
ERROR	{ Error: Authentication Error - invalid username
    at success (/var/task/node_modules/twilio/lib/base/Version.js:135:15)
    at Promise_then_fulfilled (/var/task/node_modules/q/q.js:766:44)
    at Promise_done_fulfilled (/var/task/node_modules/q/q.js:835:31)
    at Fulfilled_dispatch [as dispatch] (/var/task/node_modules/q/q.js:1229:9)
    at Pending_become_eachMessage_task (/var/task/node_modules/q/q.js:1369:30)
    at RawTask.call (/var/task/node_modules/asap/asap.js:40:19)
    at flush (/var/task/node_modules/asap/raw.js:50:29)
    at process._tickCallback (internal/process/next_tick.js:61:11)
  status: 401,
  code: 20003,
  moreInfo: 'https://www.twilio.com/docs/errors/20003',
  details: undefined }
```

**Comment by segfault** [comment]
> While testing this I did see that Twilio is no longer authenticating us which prevents us from sending SMS messages to even valid mobile numbers (tested with my mobile). Last night it was sending messages to my phone so something appears to have changed somewhere.
> 
> 
> 
> ```
> 
> ERROR	{ Error: Authentication Error - invalid username
> 
>     at success (/var/task/node_modules/twilio/lib/base/Version.js:135:15)
> 
>     at Promise_then_fulfilled (/var/task/node_modules/q/q.js:766:44)
> 
>     at Promise_done_fulfilled (/var/task/node_modules/q/q.js:835:31)
> 
>     at Fulfilled_dispatch [as dispatch] (/var/task/node_modules/q/q.js:1229:9)
> 
>     at Pending_become_eachMessage_task (/var/task/node_modules/q/q.js:1369:30)
> 
>     at RawTask.call (/var/task/node_modules/asap/asap.js:40:19)
> 
>     at flush (/var/task/node_modules/asap/raw.js:50:29)
> 
>     at process._tickCallback (internal/process/next_tick.js:61:11)
> 
>   status: 401,
> 
>   code: 20003,
> 
>   moreInfo: 'https://www.twilio.com/docs/errors/20003',
> 
>   details: undefined }
> 
> ```

I modified the keys to trigger failures and confirm the sqs behavior.

**Comment by dennisgove** [comment]
Ah, well that explains it then.

**Review by segfault — APPROVED** [review]
LGTM. 👍

**Pull request merged**